taqueria-plugin-taquito/common.ts

```diff
@@ -0,0 +1,200 @@
+import {
+	getAccountPrivateKey,
+	getDefaultAccount,
+	getNetworkConfig,
+	getSandboxAccountConfig,
+	getSandboxAccountNames,
+	getSandboxConfig,
+	sendAsyncErr,
+	sendErr,
+	TAQ_ROOT_ACCOUNT,
+} from '@taqueria/node-sdk';
+import { Environment, NetworkConfig, RequestArgs, SandboxAccountConfig, SandboxConfig } from '@taqueria/node-sdk/types';
+import { importKey, InMemorySigner } from '@taquito/signer';
+import { TezosToolkit } from '@taquito/taquito';
+
+export interface OriginateOpts extends RequestArgs.ProxyRequestArgs {
+	contract: string;
+	storage: string;
+	alias?: string;
+	sender?: string;
+}
+
+export interface TransferOpts extends RequestArgs.ProxyRequestArgs {
+	contract: string;
+	tez?: string;
+	param?: string;
+	entrypoint?: string;
+	sender?: string;
+}
+
+export interface InstantiateAccountOpts extends RequestArgs.ProxyRequestArgs {
+}
+
+export interface FundOpts extends RequestArgs.ProxyRequestArgs {
+}
+
+// To be used for the main entrypoint of the plugin
+export type IntersectionOpts = OriginateOpts & TransferOpts & InstantiateAccountOpts & FundOpts;
+
+// To be used for common functions in this file
+type UnionOpts = OriginateOpts | TransferOpts | InstantiateAccountOpts | FundOpts;
+
+export const getFirstAccountAlias = (sandboxName: string, opts: UnionOpts) => {
+	const aliases = getSandboxAccountNames(opts)(sandboxName);
+	return aliases.shift();
+};
+
+export const configureToolKitWithSandbox = async (
+	parsedArgs: UnionOpts,
+	sandboxName: string,
+	sender?: string,
+): Promise<TezosToolkit> => {
+	const sandbox = getSandboxConfig(parsedArgs)(sandboxName);
+	if (!sandbox) {
+		return sendAsyncErr(
+			`The current environment is configured to use a sandbox called '${sandboxName}'; however, no sandbox of this name has been configured in .taq/config.json.`,
+		);
+	}
+
+	let accountKey: string;
```

Comment:
I'm wondering if we should segment out the logic of finding an appropriate account to use?

Perhaps segment to a `getForgingSandboxAccount` or something like that?

Reply:
What might be better actually would be to have, `getForgingAccount` and pass in the `getSandboxAccount` or `getNetworkAccount` function as an arg.

Reply:
I'll refactor the whole thing later today btw. @mweichert